### PR TITLE
(RE-9522) Add s3 repo sync task for nightlies

### PR DESCRIPTION
**Pull request by mwaggett**
This commit adds a rake task for syncing nightlies to s3, the same way we sync apt, yum, and downloads.

### CODE REVIEW

File: tasks/ship.rake [hunk 1]
```diff
@@ -210,6 +210,17 @@ namespace :pl do
       end
     end
 
+    desc "Sync nightlies.puppetlabs.com from #{Pkg::Config.staging_server} to AWS S3"
```
Comment (shrug) [inline tasks/ship.rake]: should we use puppet.com instead of puppetlabs.com ?

Comment (mwaggett) [reply tasks/ship.rake]: We could. I was just being consistent with what we have for apt, yum, and downloads. They go to the same place.

Comment (shrug) [reply tasks/ship.rake]: Yep, I'm mostly wondering if we should make a concerted effort to update them all, or update as we touch them

Comment (mwaggett) [reply tasks/ship.rake]: I like doing them all at once, personally. because consistency++

File: tasks/ship.rake [hunk 2]
```diff
@@ -210,6 +210,17 @@ namespace :pl do
...
+    task :deploy_nightlies_to_s3 => 'pl:fetch' do
+      puts "Really run S3 sync to sync nightlies.puppetlabs.com from #{Pkg::Config.staging_server} to AWS S3? [y,n]"
+      if Pkg::Util.ask_yes_or_no
+        Pkg::Util::Execution.retry_on_fail(:times => 3) do
+          command = 'sudo /usr/local/bin/s3_repo_sync.sh nightlies.puppetlabs.com'
```
Comment (stahnma) [inline tasks/ship.rake]: Curious if we document where this comes from and how it came to be anywhere....

Comment (mwaggett) [reply tasks/ship.rake]: @stahnma https://confluence.puppetlabs.com/display/SRE/Apt+and+Yum+Download+Service+Docs

**Pull request merged**